### ISSUE TITLE
View collection doesn't get updated when WHERE clause used to check boolean field

### ISSUE THREAD

**Issue by ProgNovel** [pocketbase/pocketbase#2992]
I came across an issue I thought could be a bug.

So I have a view collection that compute the number of unread inbox of an user like this:
```sql
SELECT user as id, user, COUNT(id) as unreadCount 
FROM inbox WHERE hasOpened = false
GROUP BY user 
```

An example of `inbox` collection that it based on:
![collection ss](https://github.com/pocketbase/pocketbase/assets/45922701/fee942cd-68a6-40f5-a319-54d0732d9fea)

All records in `inbox` collection are created inside various custom "use as framework" event hooks, like when another a comment record is created by another user the custom backend will create a new `inbox` record after that and save it to the database. 

The issues I'm facing:

1. The view collection doesn't calculate the record made in custom framework, but it somehow magically picks up records that have been altered (modifying random field then hit save) or manually created records using admin UI.
2. This issue somehow doesn't appear if I remove the WHERE clause or change the condition to other value type (only tried text field type, I don't know if this also affection other types as well)
3. This one might be an unrelated issue, but I somehow can't subscribe a realtime to changes in my view collection (even though if I remove the WHERE clause). The view collection has the same API rule for both list and view operation `@request.auth.id = id`, it works when I fetch normally with `db.collection("unread_inbox_count").getOne(userId)` but `db.collection("unread_inbox_count").subscribe(userId, (e) => { ... }` doesn't give me anything.

**Comment by ganigeorgiev** [comment]
I don't understand the described issue.

Could you paste your collections configuration as json (Admin UI > Settings > Export collections) and provide more details (and ideally an example) what is the expected outcome.

**Comment by ganigeorgiev** [comment]
For 3) - view collections don't have realtime events because they don't support create/update/delete actions.

**Comment by ProgNovel** [comment]
I somehow resolve this issue on my own after I take a second look on my custom hook logic. Turns out the issue wasn't the fact that I use boolean field type in my WHERE condition, but because I use the column that still have default value in my WHERE logic.

For context, I want to make an inbox system where user is notified when someone commented on image they posted.
I have three collection; two base collection `comments` and `inbox`, and one view collection called `unread_inbox_count` that compute the number of unread inboxes (which I thought I could conveniently use realtime subscription but since it can't I might as well write some extra client-side code to watch for the `inbox` collection instead)

the collections:
[pb_schema.json.zip](https://github.com/pocketbase/pocketbase/files/12160874/pb_schema.json.zip)

I wrote custom hook as following:
```go
package events

import (
	"progstudio-app/backend/global"

	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/models"
)

func CommentCreate(e *core.RecordCreateEvent) error {
	record := e.Record

	// pb App instance is exported as global somewhere else
	imagePageRecord, err := global.App.Dao().FindRecordById("images", record.GetString("imagePage"))

	if err != nil {
		return apis.NewApiError(500, "image page record not found on comment create", nil)
	}

	imageOwnerID := imagePageRecord.GetString("user")

	// don't create inbox if both comment and the image has same owner
	if record.GetString("user") == imageOwnerID {
		return nil
	}

	commentOwnerID := record.GetString("user")
	commentOwnerRecord, err := global.App.Dao().FindRecordById("users", commentOwnerID)

	if err != nil {
		return apis.NewApiError(500, "comment's user record not found on comment create", nil)
	}

	inboxCollection, err := global.App.Dao().FindCollectionByNameOrId("inbox")
	if err != nil {
		return apis.NewApiError(500, "inbox collection not found on comment create", nil)
	}
	inboxRecord := models.NewRecord(inboxCollection)
	inboxRecord.Set("link", "/image/"+imagePageRecord.Id)
	inboxRecord.Set("user", imagePageRecord.GetString("user"))
	inboxRecord.Set("content", "@"+commentOwnerRecord.GetString("username")+" has commented on your image.")

	// I forgot to set this field which is supposed to be the condition
	// for WHERE clause. The view collection doesn't get triggered if I
	// rely on default False value from the base collection.
	//
	// Explicitely set the value here instead rely on default value
	// resolve the issue.
	inboxRecord.Set("hasOpened", false)

	global.App.Dao().SaveRecord(inboxRecord)

	return nil
}

```

**Comment by ganigeorgiev** [comment]
Thanks for the details. That's look like a bug with the default column value I'll investigate it in more details sometime later today.

**Comment by ganigeorgiev** [comment]
The issue is that for newly created records we aren't using the zero-default of the specified field when persisting the record in its table and the field value is ended up being inserted as `NULL` instead of `FALSE` in your case.

Since this isn't a blocker and you've already found a workaround, I've updated the tests and submitted a fix in the `develop` branch. It will be available with the upcoming v0.17.0 release (_most likely at the end of this week or week following it_).

Sometime later this week I'll also prepare a system migration that will normalize previously inserted `NULL` values for all non-`json` field types.

**Comment by ProgNovel** [comment]
Glad to hear that!
